amortization.go

```diff
@@ -160,9 +163,18 @@ func PlotRows(rows []Row, fileName string) error {
 		panic(err)
 	}
 	filePath := path.Join(completePath, fileName)
-	f, _ := os.Create(fmt.Sprintf("%s.html", filePath))
-	if err := bar.Render(f); err != nil {
+	f, err := os.Create(fmt.Sprintf("%s.html", filePath))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = f.Close()
+	}()
+	if writer == nil {
+		writer = f
+	}
```

Comment:
there's issue in this approach. first time file will be opened and set to `writer`. next time, same writer is being used which may not work. (file doesn't exist etc).
instead move this in unexported method. accept `io.Writer` as func param. in unit test, pass fake io.writer to this unexported method. from `PlotRows` pass real file io.Writer.

Reply:
if you find better approach, open to that as well.

Reply:
oh yes, you are right. What you suggested seems perfectly fine to me. I have added it.
There was one more approach of using https://github.com/spf13/afero. Its a fs abstraction system using which we can mock any call to the fs. But I am not inclined to use that here.